### PR TITLE
Add Clear nodes Button

### PR DESCRIPTION
**Pull request by mickr777**
Adds a Clear Nodes Button with Confirmation Dialog, I think I Did it right 😃 

I am sure there is a way to make the Confirmation look better and have Yes/No instead of OK/Cancel

### CODE REVIEW

File: invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { useTranslation } from 'react-i18next';
+import { FaTrash } from 'react-icons/fa';
+import IAIIconButton from 'common/components/IAIIconButton';
+
+const ClearNodesButton = () => {
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const handleClearNodes = () => {
+    const confirmed = window.confirm(t('common.clearNodes'));
```
Comment (blessedcoolant) [inline invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx:15]: Should use the Alert Dialog from Chakra like we have with some other components. You can reference them on how to do it.

**Review by blessedcoolant — CHANGES_REQUESTED** [review]

File: invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx (hunk 1 above)
Comment (mickr777) [reply invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx:15]: Ok will do

Comment (mickr777) [reply invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx]: @blessedcoolant I have updated it to use the AlertDialog from chakra-ui, I referenced it from the deleteboardsimagemodal, hope that was the right way

Comment (mickr777) [reply invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx]: image

File: invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx [hunk 2]
```diff
@@ -10,31 +22,68 @@ import IAIIconButton from 'common/components/IAIIconButton';
 const ClearNodesButton = () => {
   const { t } = useTranslation();
   const dispatch = useAppDispatch();
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
```
Comment (blessedcoolant) [inline invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx:25]: I think it's better to use Chakra's useDisclosure to handle the modal status. You can find documentation on that here. https://chakra-ui.com/docs/components/alert-dialog/usage

**Review by blessedcoolant — CHANGES_REQUESTED** [review]

File: invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx (hunk 2 above)
Comment (mickr777) [reply invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx]: @blessedcoolant ok changed it to useDisclosure instead of useState

**Review by blessedcoolant — APPROVED** [review]

File: invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      addToast(
+        makeToast({
+          title: t('toast.nodesCleared'),
+          status: 'success',
+        })
+      )
+    );
+
+    onClose();
+  };
```
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx:43]: Please wrap this function in a `useCallback`, for example:
```ts
  const handleConfirmClear = useCallback(() => {
    dispatch(clearNodes());

    dispatch(
      addToast(
        makeToast({
          title: t('toast.nodesCleared'),
          status: 'success',
        })
      )
    );

    onClose();
  }, [dispatch, onClose, t]);
```

Reasoning:
Every time this component re-renders, the code in the component is run again. The `handleConfirmClear` function will be recreated every time.

In simpler applications (like a website), this is no big deal, but we have a ton of stuff happening, and need to be as efficient as possible. Every millisecond counts. 

By wrapping the function in `useCallback`, we tell react to not recreate this function on every render - it's maintained with a stable reference in memory. Each render then is slightly less resource-intensive.

The dependency array `[dispatch, onClose, t]` tells react that whenever one of those things changes, it should recreate the function, so it doesn't get stale. It's important to ensure that array is correct. VSCode gives me red squigglies if it's not correct, hopefully it does for you too.

**Review by psychedelicious — CHANGES_REQUESTED** [review]
Thanks @mickr777 !

I've made two small requested changes and explained the reasoning for them.

File: invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx (hunk 3 above)
Comment (mickr777) [reply invokeai/frontend/web/src/features/nodes/components/ui/ClearNodesButton.tsx:43]: Ok, how do reopen branch thag was deleted or will i need to start a new pr

File: invokeai/frontend/web/src/features/nodes/store/nodesSlice.ts [hunk 4]
```diff
@@ -133,6 +133,10 @@ const nodesSlice = createSlice({
     loadFileEdges: (state, action: PayloadAction<Edge[]>) => {
       state.edges = action.payload;
     },
+    clearNodes: (state) => {
+      state.nodes = [];
+      state.edges = [];
+    },
```
[the remark this answers is not in the document]
Comment (blessedcoolant) [reply invokeai/frontend/web/src/features/nodes/store/nodesSlice.ts:139]: I merged the PR in already but if @mickr777 can open a new PR with these changes, I'll merge that up. If not, I'll do one myself later. Either is fine.

**Pull request merged**